engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/br/CNPJValidator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.constraintvalidators.hv.br;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.hibernate.validator.constraints.Mod11Check;
+import org.hibernate.validator.constraints.br.CNPJ;
+import org.hibernate.validator.internal.constraintvalidators.hv.Mod11CheckValidator;
+import org.hibernate.validator.internal.util.annotationfactory.AnnotationDescriptor;
+import org.hibernate.validator.internal.util.annotationfactory.AnnotationFactory;
+
+/**
+ * @author Hardy Ferentschik
+ */
+public class CNPJValidator implements ConstraintValidator<CNPJ, CharSequence> {
+	private final Mod11CheckValidator firstMod11CheckWithSeparator = new Mod11CheckValidator();
+	private final Mod11CheckValidator secondMod11CheckWithSeparator = new Mod11CheckValidator();
+
+	private final Mod11CheckValidator firstMod11CheckWithoutSeparator = new Mod11CheckValidator();
+	private final Mod11CheckValidator secondMod11CheckWithoutSeparator = new Mod11CheckValidator();
+
+	@Override
+	public void initialize(CNPJ constraintAnnotation) {
+		firstMod11CheckWithSeparator.initialize( createMod11ProxyAnnotation( 9, 14, 16, true ) );
+		secondMod11CheckWithSeparator.initialize( createMod11ProxyAnnotation( 9, 16, 17, true ) );
+
+		firstMod11CheckWithoutSeparator.initialize( createMod11ProxyAnnotation( 9, 11, 12, true ) );
+		secondMod11CheckWithoutSeparator.initialize( createMod11ProxyAnnotation( 9, 12, 13, true ) );
+	}
+
+	@Override
+	public boolean isValid(CharSequence value, ConstraintValidatorContext context) {
+		if ( value == null ) {
+			return true;
+		}
+
+		if ( value.toString().contains( "." ) ) {
```

Comment:
there can be other separators as well in the whole string, but a '.' is a given. One could change this to non digit check I guess. Remember there are still the `@Pattern` constraints of the composing constraints